src/cpp/ext/filters/census/context.cc

```diff
@@ -61,9 +59,13 @@ void GenerateClientContext(absl::string_view method, CensusContext* ctxt,
 
 size_t TraceContextSerialize(const ::opencensus::trace::SpanContext& context,
                              char* tracing_buf, size_t tracing_buf_size) {
-  GrpcTraceContext trace_ctxt(context);
-  return TraceContextEncoding::Encode(trace_ctxt, tracing_buf,
-                                      tracing_buf_size);
+  if (tracing_buf_size <
+      opencensus::trace::propagation::kGrpcTraceBinHeaderLen) {
+    return 0;
+  }
+  opencensus::trace::propagation::ToGrpcTraceBinHeader(context,
+                                                       (uint8_t*)tracing_buf);
+  return opencensus::trace::propagation::kGrpcTraceBinHeaderLen;
```

Comment:
The serialized size is never greater than this?

Reply:
Correct, the size of this encoding is fixed.